_additional/fakeClient/lib/ClientHandler.js

```diff
@@ -0,0 +1,139 @@
+const DataGenerator = require('./DataGenerator');
+const request = require('request-promise');
+
+const MESSAGE_DELAY_MIN = 1000;
+const MESSAGE_DELAY_MAX = 5000;
+
+class ClientHandler {
+	constructor(address) {
+		this._address = `http://${address}`;
+		this._isRunning = false;
+		// Next line is a trick used to print column-like messages in console.
+		// It's something like 'zero fill' with 23 characters length and spaces instead of zeros.
+		this._logPrefix = `[${address}]               `.substr(0, 23);
+		this._messageCount = 0;
+		this._nickname = DataGenerator.getNickname();
+		this._messageTimeout = null;
+		
+		this._log(`New handler created.`);
+	}
+	
+	/**
+	 * Enables fake client, registers it's username and starts the message worker.
+	 */
+	run() {
+		this._isRunning = true;
+		
+		this._register()
+			.then(this._runMessageWorker.bind(this))
+			.catch(error => {
+				this._isRunning = false;
+				this._log(error, 'error');
+			});
+	}
+	
+	/**
+	 * Stops message loop and sends http request to '/disconnect' endpoint.
+	 * @returns {Promise.<undefined>}
+	 */
+	disconnect() {
+		this._isRunning = false;
+		clearTimeout(this._messageTimeout);
+		
+		const params = {
+			method: 'POST',
+			url: `${this._address}/disconnect`,
+			form: {
+				username: this._nickname
+			}
+		};
+		
+		return request(params)
+			.then(() => this._log(`Disconnected.`))
+			.catch(this._handleError.bind(this));
+	}
+	
+	/**
+	 * Performs registration request.
+	 * @returns {Promise.<undefined>}
+	 * @private
+	 */
+	_register() {
+		const params = {
+			method: 'POST',
+			url: `${this._address}/connect`,
+			form: {
+				username: this._nickname
+			}
+		};
+		
+		return request(params)
+			.then(() => {
+				this._log(`Registered as a ${this._nickname}.`);
+			})
+			.catch(this._handleError.bind(this));
+	}
+	
+	/**
+	 * Sends pseudo-randomly generated message.
+	 * @returns {Promise.<undefined>}
+	 * @private
+	 */
+	_sendMessage() {
```

Comment:
IMO to sendMessage powinno być zdefiniowane pod _runMessageWorker.

Reply:
Dlaczego?